src/msg/async/rdma/RDMAConnectedSocketImpl.cc

```diff
@@ -233,18 +239,19 @@ void RDMAConnectedSocketImpl::handle_connection() {
         ldout(cct, 10) << __func__ << " server is already active." << dendl;
         return ;
       }
+      r = activate();
+      assert(!r);
       r = infiniband->send_msg(cct, tcp_fd, my_msg);
       if (r < 0) {
         ldout(cct, 1) << __func__ << " server ack failed." << dendl;
         dispatcher->perf_logger->inc(l_msgr_rdma_handshake_errors);
         fault();
         return ;
       }
-      r = activate();
-      assert(!r);
     } else { // ack from client
       connected = 1;
-      cleanup();
+      ldout(cct, 10) << __func__ << " handshake of rdma is done. server connected: " << connected << dendl;
+      //cleanup();
```

Comment:
CLOSE_WAIT means server side receive the FIN from client, but local side doesn't close the socket. so the right fix is close local side's socket ASAP if handshake finished. otherwise, it will wait for AsyncConnection deconstruct RDMAConnTCP.

could you try the fix to close socket in cleanup()?